src/compiler/scanner.ts

```diff
@@ -998,6 +1033,25 @@ module ts {
                     return token = SyntaxKind.EndOfFileToken;
                 }
                 let ch = text.charCodeAt(pos);
+
+                if (inJSXChild) {
+                    switch (ch) {
+                        case CharacterCodes.closeBrace:

```

Comment:
closeBrace need to be caught to disambiguate case like:
```
<div> {<div>{}} </div>
````
In speculative JSX element parsing the scanner will return an invalid closeBrace token during the parsing of the second `<div>` and `parseJSXElement` will return `null`.
see: https://github.com/fdecampredon/jsx-typescript/blob/jsx-dev/src/compiler/parser.ts#L3693-L3699